### PR TITLE
Mem map cleanup

### PR DESCRIPTION
**Pull request by jenswi-linaro**

### CODE REVIEW

File: core/arch/arm/mm/core_mmu_lpae.c [hunk 1]
```diff
@@ -609,26 +609,27 @@ static bool core_mmu_entry_copy(struct core_mmu_table_info *tbl_info,
...
-				   struct tee_mmap_region *mm)
+				   struct memory_map *mem_map)
 {
-	size_t n;
+	size_t n = 0;
 
-	assert(prtn && mm);
+	assert(prtn && mem_map);
 
-	for (n = 0; !core_mmap_is_end_of_table(mm + n); n++) {
```
Comment (jforissier) [inline core/arch/arm/mm/core_mmu_lpae.c:618]: `core_mmap_is_end_of_table()` is now unused and should be removed from `core/include/mm/core_mmu.h`, and with that `MEM_AREA_END` should be removed from `enum teecore_memtypes` as well.

**Review by jforissier — COMMENTED** [review]
- For "libutils: add array helpers": two minor comments, in any case:
  `Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`
- For "core: mm: introduce struct memory_map":
   With my comment addressed: 
  `Acked-by: Jerome Forissier <jerome.forissier@linaro.org>`
- For " core: add_phys_mem(): fix mergeable physical memory":
  `Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`

File: lib/libutils/ext/include/string_ext.h [hunk 2]
```diff
@@ -52,4 +52,66 @@ char *nex_strdup(const char *s);
...
+ * @base:        start of the array
+ * @elem_count:  the number of elements in the array
+ * @elem_size:   the size of each element in the array
+ * @pos:         element position counted in units of @elem_size
+ *
+ * Removes the element at @pos by advancing the element after this position
+ * to fill the space. The now unused element at the end of the array is
+ * zero-initialized.
+ */
+void rem_array_elem_pad_zero(void *base, size_t elem_count, size_t elem_size,
```
[the remark this answers is not in the document]
Comment (jenswi-linaro) [reply lib/libutils/ext/include/string_ext.h:114]: OK, I'll update.

File: core/arch/arm/mm/core_mmu_lpae.c (hunk 1 above)
Comment (jenswi-linaro) [reply core/arch/arm/mm/core_mmu_lpae.c:618]: Good point, I'll update.

**Comment by jenswi-linaro** [comment]
Updated and tags applied.

**Pull request merged**